File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string,int> m;
    for(auto p: participant) {
        m[p]++;
    }
    for(auto c: completion) {
        m[c]--;
    }
    vector<pair<string,int>> v(m.begin(), m.end());
    sort(v.begin(), v.end(),[](auto a, auto b){
        return a.second > b.second;
    });
    return v[0].first;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String,Integer> map = new HashMap<>();
        for(String p: participant) {
            map.put(p, map.getOrDefault(p,0) + 1);
        }
        for(String c: completion) {
            map.put(c, map.get(c)-1);
        }
        
        for(Map.Entry<String,Integer> entry: map.entrySet()) {
            if(entry.getValue() > 0) {
                return entry.getKey();
            }
        }
        
        return answer;
    }
}